src/modules/flow/http-server/http-server.c

```diff
@@ -292,6 +316,66 @@ common_close(struct sol_flow_node *node, void *data)
     struct http_data *mdata = data;
 
     stop_server(mdata);
+    free(mdata->allowed_methods);
+}
+
+static int
+parse_allowed_methods(const char *allowed_methods, enum sol_http_method **methods_array)
+{
+    struct sol_vector tokens;
+    struct sol_str_slice *token, allowed_methods_slice;
+    uint16_t i;
+    int r = -EINVAL;
+    static const struct sol_str_table map[] = {
+        SOL_STR_TABLE_PTR_ITEM("GET", SOL_HTTP_METHOD_GET),
+        SOL_STR_TABLE_PTR_ITEM("POST", SOL_HTTP_METHOD_POST),
+        { }
+    };
+
+    if (!allowed_methods) {
+        SOL_WRN("Allowed HTTP methods is NULL");
+        return -EINVAL;
+    }
+
+    allowed_methods_slice = sol_str_slice_trim(
+        sol_str_slice_from_str(allowed_methods));
+
+    if (!allowed_methods_slice.len) {
+        SOL_WRN("Allowed HTTP methods is empty");
+        return -EINVAL;
+    }
+
+    tokens = sol_str_slice_split(sol_str_slice_from_str(allowed_methods),
+        "|", 0);
+    SOL_INT_CHECK_GOTO(tokens.len, < 1, err_exit);
+
+    *methods_array = calloc(tokens.len + 1, sizeof(enum sol_http_method));
+    SOL_NULL_CHECK_GOTO(*methods_array, err_nomem_exit);
```

Comment:
argh... I don't like the split and this array, adds memory allocations for no good reason. Parse using strchr and slices, then convert to enum and then use a bitmask. Somehow like:

```c
uint8_t allowed = 0;
while (*allowed_methods) {
    const char *ptr = strchr(allowed_methods, '|');
    struct sol_str_slice slice;

    if (!ptr) ptr = allowed_methods + strlen(allowed_methods);
    slice.data = allowed_methods;
    slice.len = ptr - allowed_methods;

    if (sol_str_slice_eq(slice, "GET"))
        allowed |= (1 << SOL_HTTP_METHOD_GET);
    else if (sol_str_slice_eq(slice, "POST"))
        allowed |= (1 << SOL_HTTP_METHOD_POST);
    else SOL_WRN("Unsupported allowed_method=%.*s", SOL_STR_SLICE_PRINT(slice));

    if (*ptr == '|')
       allowed_methods = endptr + 1;
   else break;
}
```

Then checks are:

```c
if (mdata->allowed & (1 << method)) { ...
```